master/internal/api/api_test.go

```diff
@@ -0,0 +1,51 @@
+package api
+
+import (
+	"testing"
+)
+
+func TestPaginate(t *testing.T) {
+	tables := []struct {
+		Total       int
+		Offset      int
+		Limit       int
+		ReturnCount int
+		Err         bool
+	}{
+		{10, 0, 0, 10, false},
+		{10, 1, 0, 9, false},
+		{10, 1, 1, 1, false},
+		{10, -1, 0, 1, false},
+		{10, -2, 1, 1, false},
+		{10, 13, 0, 7, true},
```

Comment:
nit: might have added startIndex & endIndex to the values we verify

Reply:
I added startIndex. the end should be indirectly tested by startIndex & total.